delfin/drivers/dell_emc/vplex/vplex_stor.py

```diff
@@ -324,16 +324,221 @@ def get_cluster_used_capacity(self, cluster_name):
+
+    @staticmethod
+    def analyse_director_status(status):
+        return consts.CONTROLLER_STATUS_MAP. \
+            get(status, constants.ControllerStatus.UNKNOWN)
+
+    def get_director_specified_version(self, version_map, director_name,
+                                       specified_name):
+        version_value = ''
+        if version_map:
+            director_map = version_map.get(director_name)
+            if director_map:
+                version_value = director_map.get(specified_name)
+        return version_value
+
+    def get_value_from_nest_map(self, nest_map, first_key, second_key):
+        final_value = ''
+        if nest_map:
+            second_map = nest_map.get(first_key)
+            if second_map:
+                final_value = second_map.get(second_key)
+        return final_value
+
+    def get_hardware_port_info(self, nest_map, first_key, second_key):
+        speed = ''
+        max_speed = ''
+        protocols = []
+        role = ''
+        port_status = ''
+        operational_status = ''
+
+        if nest_map:
+            second_map = nest_map.get(first_key)
+            if second_map:
+                third_map = second_map.get(second_key)
+                if third_map:
+                    speed = third_map.get('current-speed')
+                    max_speed = third_map.get('max-speed')
+                    protocols = third_map.get('protocols')
+                    role = third_map.get('role')
+                    port_status = third_map.get('port-status')
+                    operational_status = third_map.get('operational-status')
+        return (speed, max_speed, protocols, role, port_status,
+                operational_status)
+
+    @staticmethod
+    def analyse_hardware_port(resp, hardware_port_map):
+        port_list = VplexStorageDriver.get_context_list(resp)
+        if port_list:
+            for port in port_list:
+                port_name = port.get("attributes").get("target-port")
+                hardware_port_map[port_name] = port
+
+    @staticmethod
+    def analyse_port_type(protocols):
+        port_type = constants.PortType.OTHER
+        if protocols:
+            for protocol in protocols:
+                port_type_value = consts.PORT_TYPE_MAP.get(protocol)
+                if port_type_value:
+                    port_type = port_type_value
+                    break
+        return port_type
+
+    @staticmethod
+    def analyse_port_logical_type(role):
+        return consts.PORT_LOGICAL_TYPE_MAP. \
+            get(role, constants.PortLogicalType.OTHER)
+
+    @staticmethod
+    def analyse_port_connect_status(status):
+        return consts.PORT_CONNECT_STATUS_MAP. \
+            get(status, constants.PortConnectionStatus.UNKNOWN)
+
+    @staticmethod
+    def analyse_port_health_status(status):
+        return consts.PORT_HEALTH_STATUS_MAP. \
+            get(status, constants.PortHealthStatus.UNKNOWN)
+
+    @staticmethod
+    def analyse_speed(speed):
```

Comment:
What's this function for? It seems like something about speed, but it used `units.G`, which is a unit for capacity.

Reply:
Modified